### PR TITLE
Added  *fetch-through-merge-base

### PR DESCRIPTION
**Pull request by michal-dagan**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: [link to the issue
](https://jira-dc.paloaltonetworks.com/browse/CRTX-92566)
## Description
Increased GIT_DEPTH from 0 to 50 and added the [fetch_through_merge_base](https://github.com/rmacklin/fetch-through-merge-base/blob/main/fetch_through_merge_base.sh).

**Trigger Content Nightly:** https://code.pan.run/xsoar/content/-/pipelines/6896434
new: https://code.pan.run/xsoar/content/-/pipelines/6907774

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

File: .gitlab/ci/.gitlab-ci.variables.yml [hunk 1]
```diff
@@ -26,7 +26,7 @@ variables:
   PIP_CACHE_DIR: "$CI_PROJECT_DIR/.cache/pip"
   FEATURE_BRANCH_NAME: "v4.5.0"
   IFRA_ENV_TYPE: "Content-Env"
-  GIT_DEPTH: 0
+  GIT_DEPTH: 1
```
Comment (michal-dagan) [inline .gitlab/ci/.gitlab-ci.variables.yml:29]:
```suggestion
  GIT_DEPTH: 100
```

File: .gitlab/ci/.gitlab-ci.variables.yml [hunk 2]
```diff
@@ -26,7 +26,7 @@ variables:
   PIP_CACHE_DIR: "$CI_PROJECT_DIR/.cache/pip"
   FEATURE_BRANCH_NAME: "v4.5.0"
   IFRA_ENV_TYPE: "Content-Env"
-  GIT_DEPTH: 0
+  GIT_DEPTH: 100
```
Comment (michal-dagan) [inline .gitlab/ci/.gitlab-ci.variables.yml:29]:
```suggestion
  GIT_DEPTH: 1000
```

File: .gitlab/ci/.gitlab-ci.global.yml [hunk 3]
```diff
@@ -265,6 +265,14 @@
     - *build_parameters
     - *gitlab_ci_build_parameters
     - *checkout-upload-commit-content-nightly
+    - DEEPEN_LENGTH=${DEEPEN_LENGTH:-50}
+    - HEAD_REF=origin/master
+    - BASE_REF=$(git rev-parse HEAD)
+    - |
+      while [ -z "$( git merge-base "$BASE_REF" "$HEAD_REF" )" ]; do
+        echo "Fetching with depth $DEEPEN_LENGTH..."
+        git fetch -q --deepen="$DEEPEN_LENGTH" origin "$BASE_REF" "$HEAD_REF";
```
Comment (michal-dagan) [inline .gitlab/ci/.gitlab-ci.global.yml:274]:
```suggestion
        git fetch -q --deepen="$DEEPEN_LENGTH" origin;
```

File: .gitlab/ci/.gitlab-ci.global.yml [hunk 4]
```diff
@@ -246,6 +246,15 @@
   - |
   - section_end "Checkout upload commit content nightly"
 
+.fetch-through-merge-base: &fetch-through-merge-base
+  - DEEPEN_LENGTH=${DEEPEN_LENGTH:-50}
```
Comment (kobymeir) [inline .gitlab/ci/.gitlab-ci.global.yml:250]: don't run it when the current branch is master

**Review by kobymeir — CHANGES_REQUESTED** [review]

**Review by kobymeir — APPROVED** [review]

**Pull request merged**